tests/container-copy

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+set -eu
+
+# Install and setup LXD.
+install_lxd
+lxd init --auto
+
+# Create a loop device that we can mount as a source for a dir storage pool.
+test_dir="$(mktemp -d XXX.test)"
+loopimg="$(mktemp "${test_dir}/XXX.img")"
+truncate -s 5G "${loopimg}"
+loopdev="$(losetup --show --find "${loopimg}")"
+
+# Begin Test.
+set -x
+
+# Launch a VM which will be the target of the copy.
+lxc launch "${TEST_IMG:-ubuntu-minimal-daily:22.04}" target --vm
+waitInstanceBooted target
+
+# Add the target VM as a remote LXD.
+target_addr="$(lxc ls -f csv -c 4 target | grep -v lxdbr | cut -d' ' -f1 | sed 's/"//g')"
+lxc exec target -- snap install lxd --channel "${LXD_SNAP_CHANNEL}"
+lxc exec target -- lxd init --auto
+lxc exec target -- lxc config set core.https_address=:8443
+lxc exec target -- lxc config set core.trust_password=lxd123
+echo "lxd123" | lxc remote add target "https://${target_addr}:8443" --accept-certificate
+
+# Run the test for various filesystems.
+for fs in "xfs" "btrfs" "ext4" ; do
+  mkfs.${fs} "${loopdev}"
+  disk_dir="$(mktemp -d "${test_dir}/XXX.disk")"
+  mount "${loopdev}" "${disk_dir}"
+  rm -rf "${disk_dir}/lost+found"
+
+  # Create a storage pool with the mounted disk.
+  lxc storage create dir dir source="${PWD}/${disk_dir}"
```

Comment:
To ease debugging, it would help to embed the FS name in the storage pool name, please.